### PR TITLE
Feature/delete answer

### PR DESCRIPTION
**Pull request by suryasinghvijay**

### CODE REVIEW

File: quora-service/src/main/java/com/upgrad/quora/service/entity/UserEntity.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
                 @NamedQuery(name = "getUserByEmail", query = "select u from UserEntity u where u.email=:email"),
                 @NamedQuery(name = "getUserByUserName", query = "select u from UserEntity u where u.userName=:username"),
                 @NamedQuery(name = "getUserByUserId", query = "select u from UserEntity u where u.uuid=:uuid"),
-                @NamedQuery(name = "deleteUserByUserId", query = "DELETE FROM UserEntity c WHERE c.uuid = :uuid")
+                @NamedQuery(name = "deleteUserByUserId", query = "DELETE FROM UserEntity  WHERE uuid = :uuid")
```
Comment (arunsubashm) [inline quora-service/src/main/java/com/upgrad/quora/service/entity/UserEntity.java]: Why is the change needed

File: quora-service/src/main/java/com/upgrad/quora/service/entity/AnswerEntity.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    @Column(name = "date")
+    @NotNull
+    private ZonedDateTime date;
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "user_id")
+    @OnDelete(action = OnDeleteAction.CASCADE)
+    private UserEntity userId;
+
+    @ManyToOne(fetch = FetchType.LAZY)
```
Comment (arunsubashm) [inline quora-service/src/main/java/com/upgrad/quora/service/entity/AnswerEntity.java:46]: Why is it LAZY

File: quora-api/src/main/java/com/upgrad/quora/api/controller/AnswerController.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+     * @return
+     * @throws AuthorizationFailedException
+     * @throws UserNotFoundException
+     * @throws AnswerNotFoundException
+     */
+    @RequestMapping(method = RequestMethod.DELETE, path = "/answer/delete/{answerId}", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<AnswerDeleteResponse> deleteUser(@RequestHeader("authorization") final String authorization, @PathVariable String answerId) throws AuthorizationFailedException,
+            UserNotFoundException, AnswerNotFoundException {
+
+        final UserAuthTokenEntity userAuthDetails = authenticationService.getAuthToken(authorization, "User has not signed in", "User is signed out.Sign in first to post an answer");
```
Comment (arunsubashm) [inline quora-api/src/main/java/com/upgrad/quora/api/controller/AnswerController.java]: Can all this be moved to the service function

**Review by arunsubashm — CHANGES_REQUESTED** [review]

File: quora-service/src/main/java/com/upgrad/quora/service/entity/AnswerEntity.java (hunk 2 above)
Comment (suryasinghvijay) [reply quora-service/src/main/java/com/upgrad/quora/service/entity/AnswerEntity.java:46]: to defer initialization of an object as long as it's possible

Comment (arunsubashm) [reply quora-service/src/main/java/com/upgrad/quora/service/entity/AnswerEntity.java:46]: Thanks for explaining

File: quora-service/src/main/java/com/upgrad/quora/service/entity/UserEntity.java (hunk 1 above)
Comment (suryasinghvijay) [reply quora-service/src/main/java/com/upgrad/quora/service/entity/UserEntity.java]: reverted

Comment (arunsubashm) [reply quora-service/src/main/java/com/upgrad/quora/service/entity/UserEntity.java]: Fine

**Review by arunsubashm — APPROVED** [review]

**Comment by arunsubashm** [comment]
Changes look good. Thanks for taking care of the comments

**Pull request merged**